aten/src/ATen/core/jit_type.h

```diff
@@ -1476,6 +1476,11 @@ struct CAFFE2_API ClassType : public NamedType {
       TypePtr type,
       bool is_parameter = false);
 
+  // [Internal Only] Remove attribute from the ClassType
+  // caller is responsible to make sure the attribute is
+  // no longer used
+  void removeAttribute(const std::string& name);
```

Comment:
Let's put the warning in the function name: `unsafeRemoveAttribute`. Let's also add what it means for the attribute to be no longer used: it is unsafe  to having existing allocations of this object around anymore, and any code that works on the attribute is now invalid. Only newly created code is valid again.